src/vecs/collection.py

```diff
@@ -195,6 +198,10 @@ def __init__(
                 "Dimensions reported by adapter, dimension, and collection do not match"
             )
 
+        with self.client.Session() as sess:
+            with sess.begin():
+                sess.execute(text(f"create schema if not exists {self.schema};"))
```

Comment:
with the schema being user-defined all string interpolation needs to be safely sanitized. iirc is something like 
```
engine.dialect.identifier_preparer.quote(schema_name)
```
otherwise, as written, there will be no distinction between upper and lowercase letters (postgres will coerce all of them to lower case).

How about a property on the `Collection` that can be accessed to get the safely quoted name using the method ^?

Reply:
updated - I used  the identifier_preparer from `self.client.engine`